lib/deaf_grandma.rb

```diff
@@ -14,13 +14,27 @@ def run!
     loop do
       user_input = get_user_input
       p speak(user_input)
+      if @bye_counter > 0
+        exit(0)
+      end
     end
   end
 
 
   def speak(input)
 
-    #Implement your code here <<<<<<<<<
+    response = "SPEAK UP SONNY!"
+
+    if input.upcase!.nil?
```

Comment:
@gwadej If you mutate an empty string `.upcase!` instead of returning a copy `.upcase` it returns nil...interesting

```
irb(main):001:0> "".upcase
=> ""
irb(main):002:0> "".upcase!
=> nil
irb(main):003:0>
```